Intermittent build failure: json5 timing test

### Steps To Reproduce

Steps to reproduce the behavior:

1. `nix-build --attr python3Packages.json5 --check`

### Build log

https://gist.github.com/l0b0/7eb402a75380b6415f904b92d0f7de6e

### Additional context

I don't have the time to test against other Python versions. Presumably it should be enough to skip `tests/lib_test.py::TestDumps::test_object_roundtrip` and any other tests using `hypothesis.errors.Flaky`.

### Notify maintainers

@veehaitch 

### Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.41, NixOS, 23.05 (Stoat), 23.05.2274.6dc93f0daec`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"nixos-23.05"`
 - channels(username): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

I just got this from trying to build my home-manager config.


<details>
```
=========================== short test summary info ============================
FAILED tests/lib_test.py::TestDumps::test_object_roundtrip - hypothesis.errors.Flaky: Hypothesis test_object_roundtrip(self=<tests.lib_t...
======================== 1 failed, 67 passed in 22.68s =========================
test.py::test_encoding_detection[tests/windows-1255-hebrew/halemo.net.edoar.xml-windows-1255] PASSED [ 11%]
test.py::test_encoding_detection[tests/windows-1255-hebrew/hagada.org.il.xml-windows-1255] PASSED [ 11%]
test.py::test_encoding_detection[tests/windows-1255-hebrew/notes.co.il.50.xml-windows-1255] PASSED [ 11%]
/nix/store/c9bk0v8vlqvf6n126rvnq22r906zy33h-stdenv-darwin/setup: line 1587: pop_var_context: head of shell_variables not a function context
error: builder for '/nix/store/lsh1l9l7cxhxjzsqicx0alc2vrr15451-python3.10-json5-0.9.14.drv' failed with exit code 1;
       last 25 log lines:
       > E                  None,
       > E                  '1lzoiE%<me\t',
       > E                  None,
       > E                  None,
       > E                  inf,
       > E                  -1415090511218434.0],
       > E                 {'Z9[a;N': True},
       > E                 [1.9]],
       > E                '6?%a5[H:\tKX?k\nYR8': [[False, False, None, None],
       > E                 {'ZZ3%/AJb': None},
       > E                 [None]],
       > E                '17t[H,z\n,\\2$(A': [[-564049532618056.0, 1.9, False]],
       > E                'N': {'': None,
       > E                 '661mMF[jAN4)b>KdG7E[': 9007199254740992.0,
       > E                 '8J3d': '\rQ7\t1G',
       > E                 '9cMjC~2\x0bwdKHy$\\4m!': '',
       > E                 "AA;\\0$\rl{7']f/G@wuZxh": True}},
       > E           )
       > E           Unreliable test timings! On an initial run, this test took 299.72ms, which exceeded the deadline of 200.00ms, but on a subsequent run it took 19.69 ms, which did not. If you expect this sort of variability in your test timings, consider turning deadlines off for this test by setting deadline=None.
       >
       > /nix/store/n7j58wgl56kwfbga870033h9ipmic66g-python3.10-hypothesis-6.98.17/lib/python3.10/site-packages/hypothesis/core.py:971: Flaky
       > =========================== short test summary info ============================
       > FAILED tests/lib_test.py::TestDumps::test_object_roundtrip - hypothesis.errors.Flaky: Hypothesis test_object_roundtrip(self=<tests.lib_t...
       > ======================== 1 failed, 67 passed in 22.68s =========================
       > /nix/store/c9bk0v8vlqvf6n126rvnq22r906zy33h-stdenv-darwin/setup: line 1587: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/lsh1l9l7cxhxjzsqicx0alc2vrr15451-python3.10-json5-0.9.14.drv'.
error: 1 dependencies of derivation '/nix/store/lbjk66sh2mwm7661phfa4aqlan3ac3mk-python3-3.10.14-env.drv' failed to build
error (ignored): error: cannot unlink '/private/tmp/nix-build-tree-sitter-0.20.9.drv-0/source/target/aarch64-apple-darwin/release/deps': Directory not empty
error: 1 dependencies of derivation '/nix/store/7cjma610s7csv67dm6sh1wphjp9fjcm9-home-manager-applications.drv' failed to build
error: 1 dependencies of derivation '/nix/store/lfaf5yw1cw0h75y7gwa39m37q8q0zng6-home-manager-fonts.drv' failed to build
error: 1 dependencies of derivation '/nix/store/8mc32y5xi0hqdgd836bxpq84m1ykl1y9-home-manager-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/px5cgg8vgrw9qhagg3gndb66jlrf2p88-home-manager-generation.drv' failed to build
```
</details>

--- a/pkgs/development/python-modules/json5/default.nix
+++ b/pkgs/development/python-modules/json5/default.nix
@@ -1,6 +1,5 @@
 { buildPythonPackage
 , fetchFromGitHub
-, hypothesis
 , lib
 , pytestCheckHook
 }:
@@ -18,7 +17,6 @@ buildPythonPackage rec {
   };
 
   nativeCheckInputs = [
-    hypothesis
     pytestCheckHook
   ];
 

